In [1]:
import pandas as pd
import numpy as np
import math

# Read in the data
df1 = pd.read_csv('./Data/HPO_input/Results/GPT4_top_prediction_RE.csv')
# split columns into multiple columns
df1['dataset'] = df1['file_path'].str.split('/', expand=True)[4]
df2 = pd.read_csv('./Data/simulated_pt_input/Results/GPT4_top_prediction_RE.csv')
df2['dataset'] = df2['file_path'].str.split('/', expand=True)[4]
df1['is_simulation'] = False
df2['is_simulation'] = True

In [2]:
# format float to 2 decimal places
def prediction_accuracy_summary(df):
    w = df.shape[0]
    x = df[['predict_correct_in_top_5']].values.sum()
    y = df[['predict_made_in_top_10']].values.sum()
    print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

    x = df[['predict_correct_in_top_10']].values.sum()
    y = df[['predict_made_in_top_10']].values.sum()
    print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

    x = df[['predict_correct_in_top_50']].values.sum()
    y = df[['predict_made_in_top_50']].values.sum()
    print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
    print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

def prediction_accuracy_summary_by_subset(df):
    # subset df by dataset
    dataset_list = df['dataset'].unique()
    for dataset in dataset_list:
        print(dataset)
        df_subset = df[df['dataset']==dataset]
        w = df_subset.shape[0]
        x = df_subset[['predict_correct_in_top_5']].values.sum()
        y = df_subset[['predict_made_in_top_5']].values.sum()
        print('{y} out of {w} predictions made in top 5 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
        print('{x} out of {y} predictions correct in top 5 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

        x = df_subset[['predict_correct_in_top_10']].values.sum()
        y = df_subset[['predict_made_in_top_10']].values.sum()
        print('{y} out of {w} predictions made in top 10 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
        print('{x} out of {y} predictions correct in top 10 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

        x = df_subset[['predict_correct_in_top_50']].values.sum()
        y = df_subset[['predict_made_in_top_50']].values.sum()
        print('{y} out of {w} predictions made in top 50 ({z}%)'.format(y=y, w=w, z=np.round(y/w,4)*100))
        print('{x} out of {y} predictions correct in top 50 ({z}%)'.format(x=x, y=y, z=np.round(x/y,4)*100))

In [3]:
for df in [df1, df2]:
    print(f"Is it a simulation dataset: {df['is_simulation'].tolist()[0]}")
    prediction_accuracy_summary(df)
    prediction_accuracy_summary_by_subset(df)


Is it a simulation dataset: False
266 out of 276 predictions made in top 5 (96.38%)
28 out of 266 predictions correct in top 5 (10.530000000000001%)
266 out of 276 predictions made in top 10 (96.38%)
36 out of 266 predictions correct in top 10 (13.530000000000001%)
206 out of 276 predictions made in top 50 (74.64%)
35 out of 206 predictions correct in top 50 (16.99%)
ColumbiaU
27 out of 27 predictions made in top 5 (100.0%)
6 out of 27 predictions correct in top 5 (22.220000000000002%)
27 out of 27 predictions made in top 10 (100.0%)
7 out of 27 predictions correct in top 10 (25.929999999999996%)
22 out of 27 predictions made in top 50 (81.47999999999999%)
6 out of 22 predictions correct in top 50 (27.27%)
CSH
72 out of 72 predictions made in top 5 (100.0%)
12 out of 72 predictions correct in top 5 (16.669999999999998%)
68 out of 72 predictions made in top 10 (94.44%)
14 out of 68 predictions correct in top 10 (20.59%)
53 out of 72 predictions made in top 50 (73.61%)
15 out of 53 predi

In [51]:
def probability_of_picking_specific_card(total_cards, specific_cards, draws):
    probability_not_picking = 1

    for i in range(draws):
        non_specific_cards = total_cards - specific_cards - i
        remaining_cards = total_cards - i
        probability_not_picking *= non_specific_cards / remaining_cards

    probability_picking = 1 - probability_not_picking
    return probability_picking
# calculate the probability of select the correct gene in top 5

total_cards = 30000
specific_cards = 1
# what is the probability of selecting the correct gene in top 5
draws = 5
probability = probability_of_picking_specific_card(total_cards, specific_cards, draws)
print('Probability of selecting the correct gene in top 5: {}%'.format(probability*100))
draws = 10
probability = probability_of_picking_specific_card(total_cards, specific_cards, draws)
print('Probability of selecting the correct gene in top 10: {}%'.format(probability*100))
draws = 50
probability = probability_of_picking_specific_card(total_cards, specific_cards, draws)
print('Probability of selecting the correct gene in top 50: {}%'.format(probability*100))

Probability of selecting the correct gene in top 5: 0.01666666666666483%
Probability of selecting the correct gene in top 10: 0.033333333333340764%
Probability of selecting the correct gene in top 50: 0.16666666666667052%


In [52]:
# This table is generated using ChatGPT-3

| Dataset | Top 5 Predictions | Top 5 Correct Predictions | Top 10 Predictions | Top 10 Correct Predictions | Top 50 Predictions | Top 50 Correct Predictions |
|---------|------------------|--------------------------|--------------------|----------------------------|--------------------|----------------------------|
| Real HPO Input|
|         | 266 (96.38%)     | 28 (10.53%)              | 266 (96.38%)       | 36 (13.53%)                | 206 (74.64%)       | 35 (16.99%)                |
| ColumbiaU | 27 (100.0%)      | 6 (22.22%)               | 27 (100.0%)        | 7 (25.93%)                 | 22 (81.48%)        | 6 (27.27%)                 |
| CSH     | 72 (100.0%)      | 12 (16.67%)              | 68 (94.44%)        | 14 (20.59%)                | 53 (73.61%)        | 15 (28.30%)                |
| AJHG    | 78 (100.0%)      | 0 (0.0%)                  | 76 (97.44%)        | 0 (0.0%)                   | 52 (66.67%)        | 1 (1.92%)                  |
| DGD     | 85 (100.0%)      | 10 (11.76%)              | 82 (96.47%)        | 15 (18.29%)                | 67 (78.82%)        | 13 (19.40%)                |
| TAF1    | 14 (100.0%)      | 0 (0.0%)                  | 13 (92.86%)        | 0 (0.0%)                   | 12 (85.71%)        | 0 (0.0%)                   |
| Simulated HPO Input    |                            |
|         | 274 (99.28%)     | 26 (9.49%)               | 274 (99.28%)       | 32 (11.68%)                | 208 (75.36%)       | 45 (21.63%)                |
| ColumbiaU | 27 (100.0%)      | 5 (18.52%)               | 27 (100.0%)        | 6 (22.22%)                 | 23 (85.19%)        | 7 (30.43%)                 |
| CSH     | 72 (100.0%)      | 11 (15.28%)              | 72 (100.0%)        | 13 (18.06%)                | 56 (77.78%)        | 19 (33.93%)                |
| AJHG    | 78 (100.0%)      | 0 (0.0%)                  | 78 (100.0%)        | 0 (0.0%)                   | 48 (61.54%)        | 0 (0.0%)                   |
| DGD     | 85 (100.0%)      | 10 (12.20%)              | 83 (97.65%)        | 13 (15.66%)                | 72 (84.71%)        | 19 (26.39%)                |
| TAF1    | 13 (92.86%)      | 0 (0.0%)                  | 14 (100.0%)        | 0 (0.0%)                   | 9 (64.29%)         | 0 (0.0%)                   |

